src/frontend/pages/signup/signup.js

```diff
@@ -0,0 +1,132 @@
+import { useAuth } from "../../contexts/authContext";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import "./signup.css";
+
+function Signup() {
+  const { authState, authDispatch } = useAuth();
+  const { user } = authState;
+
+  const navigate = useNavigate();
+
+  const formSignupHandler = async (e) => {
+    e.preventDefault();
+    console.log(e);
+    try {
+      const response = await axios.post("/api/auth/signup", {
+        firstName: user.firstName,
+        lastName: user.lastName,
+        email: user.email,
+        password: user.password,
+      });
+      console.log(user.firstName);
+      localStorage.setItem("token", response.data.encodedToken);
+      authDispatch({
+        authType: "TOKEN",
+        authPayload: response.data.encodedToken,
+      });
+      navigate("/");
+    } catch (error) {
+      authDispatch({ authType: "ERROR", authPayload: true });
+      console.log(`Error: ${error}`);
+    }
+  };
+
+  return (
+    <div className="center">
+      <form
+        className="input-container ecom-placing-input-container"
+        onSubmit={formSignupHandler}
+      >
+        <h2 className="ecom-center-inside-flex">CREATE ACCOUNT</h2>
+        <div className="input-name">
+          <label for="firstName">FIRST NAME:</label>
```

Comment:
use htmlFor here instead for as react ignores that

Reply:
did not knew this. thanks will change.